File: Assignment_50X/Assignment_50.5.cpp
#include <iostream>
using namespace std;

struct node
{
    int data;
    struct node *next;
};

typedef struct node NODE;
typedef struct node *PNODE;

class SinglyLL
{
public:
    PNODE first;
    int iCount;

public:
    SinglyLL()
    {
        cout << "Object of SinglyLL gets created" << "\n";
        first = NULL;
        iCount = 0;
    }

    void InsertFirst(int no)
    {
        PNODE newn = NULL;

        newn = new NODE;

        newn->data = no;
        newn->next = NULL;

        if (first == NULL)
        {
            first = newn;
        }
        else
        {
            newn->next = first;
            first = newn;
        }
    }

    void Display()
    {
        PNODE temp = NULL;

        temp = first;

        while (temp != NULL)
        {
            cout << "| " << temp->data << " |->";
            temp = temp->next;
        }

        cout << "NULL\n";
    }

    int DisplayCount(int pos)
    {
        PNODE temp = NULL;

        temp = first;

        int Count = 0;
        int iPos = 1;

        while (temp != NULL)
        {
            if((iPos % pos) == 0)
            {
                Count++;
            }
            iPos++;
            temp = temp->next;
        }
        return Count;
    }
};

int main()
{
    SinglyLL obj;
    int iRet = 0;

    obj.InsertFirst(51);
    obj.InsertFirst(40);
    obj.InsertFirst(31);
    obj.InsertFirst(20);
    obj.InsertFirst(-11);
    obj.InsertFirst(10);

    obj.Display();

    iRet = obj.DisplayCount(2);

    cout<<"Count Of Node At N th Position is: "<<iRet;

    return 0;
}

File: Assignment_50XX/Assignment_50_5.java


class node
{
    public int data;
    public node next;
    
    public node(int no)
    {
        data = no;
        next = null;
    }
}

class SinglyLL
{
    private node first;
    private int iCount;

    public SinglyLL() 
    {
        System.out.println("Object Of SinglyLL gets created");
        first = null;
        iCount = 0;
    }

    public void InsertFirst(int no)
    {
        node newn  = null;

        newn = new node(no);

        if(first == null)
        {
            first = newn;
        }
        else
        {
            newn.next = first;
            first = newn;
        }
    }
    public void Display()
    {
        node temp = null;

        temp = first;

        while(temp != null)
        {
            System.out.print("| "+temp.data+" |->");
            temp = temp.next;
        }

        System.out.println("null\n");
    }

    public int DisplayCount(int iPosition)
    {       
        node temp = null;

        temp = first; 
        int pos = 1;

        while(temp != null)
        {   
            if((pos == iPosition))
            {   
               return 1;
            }
            pos++;
            temp = temp.next;   
        } 
        return 0;
    }
}

class Assignment_50_5
{
    public static void main(String Arr[])
    {
        SinglyLL obj;

      int iRet = 0; 

        obj = new SinglyLL();

        obj.InsertFirst(51);
        obj.InsertFirst(41);
        obj.InsertFirst(20);
        obj.InsertFirst(25);
        obj.InsertFirst(10);
        obj.InsertFirst(55);

        obj.Display();

        iRet = obj.DisplayCount(2);

            System.out.println("Count Of Node At N th Position is:"+iRet);
    
        
        
        obj = null;

        System.gc();
    }
}